### PR TITLE
Auth:  refresh token handling and test cases

### PR DESCRIPTION
**Pull request by bmccandless**
Part of issue chanzuckerberg/corpora-data-portal#481

### CODE REVIEW

File: backend/corpora/common/authorizer.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-from jose import jwt, JWTError
-from .corpora_config import CorporaAuthConfig
+from backend.corpora.common.corpora_config import CorporaAuthConfig
+from jose import jwt
+from jose.exceptions import ExpiredSignatureError, JWTError, JWTClaimsError
 
 
 def assert_authorized_token(token: str) -> dict:
-    """
+    """ "
```
Comment (Bento007) [inline backend/corpora/common/authorizer.py:11]: I think you have a stray

Comment (Bento007) [reply backend/corpora/common/authorizer.py:11]:
```suggestion
    """
```

Comment (bmccandless) [reply backend/corpora/common/authorizer.py]: fixed

**Review by tihuan — COMMENTED** [review]
Looks like this is super BE stuff, I don't have much contribution here 😆 LGTM!

File: tests/unit/backend/chalice/api_server/test_v1_auth.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+# function to launch the mock oauth server
+def launch_mock_oauth():
+    mock_oauth_app.run(port=PORT)
+
+
+@unittest.skipIf(
+    os.environ["DEPLOYMENT_STAGE"] != "test",
+    f"Does not run DEPLOYMENT_STAGE:{os.environ['DEPLOYMENT_STAGE']}",
+)
+class TestAuth(unittest.TestCase):
```
Comment (Bento007) [inline tests/unit/backend/chalice/api_server/test_v1_auth.py:80]: inherit  https://github.com/chanzuckerberg/corpora-data-portal/blob/7eb024c2297255901f053cc9b61a241a2de848b3/tests/unit/backend/chalice/api_server/__init__.py#L8

File: tests/unit/backend/chalice/api_server/test_v1_auth.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    def check_user_info(self, userinfo):
+        self.assertEqual(userinfo["is_authenticated"], True)
+        self.assertEqual(userinfo["id"], "fake_id")
+        self.assertEqual(userinfo["name"], "Fake User")
+        self.assertEqual(userinfo["email"], "fake_user@email.com")
+        self.assertEqual(userinfo["email_verified"], True)
+
+    def test__auth_flow(self):
+        corpora_api_dir = os.path.join(os.environ["CORPORA_HOME"], "backend", "chalice", "api_server")
+        self.app = ChaliceTestHarness(corpora_api_dir)
```
Comment (Bento007) [inline tests/unit/backend/chalice/api_server/test_v1_auth.py:97]: use https://github.com/chanzuckerberg/corpora-data-portal/blob/7eb024c2297255901f053cc9b61a241a2de848b3/tests/unit/backend/chalice/api_server/__init__.py#L16

**Review by Bento007 — CHANGES_REQUESTED** [review]
Functionally it looks sounds. Some touch ups are still needed.

Follow ReStructuredText for docstring, for example https://queirozf.com/entries/python-docstrings-reference-examples#restructuredtext-rest. Don't worry about including return type. That will be iin the python typing.

File: backend/corpora/common/corpora_config.py [hunk 4]
```diff
@@ -18,6 +18,28 @@ def __init__(self, *args, **kwargs):
 
 
 class CorporaAuthConfig(SecretConfig):
+
+    """
+    secret keys:
```
[the remark this answers is not in the document]
Comment (bmccandless) [reply backend/corpora/common/corpora_config.py:23]: I wasn't sure what you meant.  Was it just to make a json files that looks like the comment here (key and description of the key), then link that file into the readme in that table with other links?

[the remark this answers is not in the document]
Comment (Bento007) [reply backend/corpora/common/corpora_config.py:23]: yes exactly

[the remark this answers is not in the document]
Comment (Bento007) [reply backend/corpora/common/corpora_config.py:23]: otherwise the format for this file will be hidden in the code.

**Review by Bento007 — APPROVED** [review]
LGTM

**Pull request merged**